core/src/common.cpp

```diff
@@ -362,6 +364,434 @@ namespace irods::http
+			else if (algorithm_specifics == "512") {
+				logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+				_verifier.allow_algorithm(jwt::algorithm::hs512(key));
+			}
+			return;
+		}
+
+		logging::warn("{}: Algorithm [{}] is not supported.", __func__, _alg);
+	}
+
+	/// Adds the specified asymmetric algorithm \p _alg to the verifier \p _verifier, using the additional information
+	/// provided by the JWK \p _jwk.
+	///
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
+	///
+	/// \param[in,out] _verifier The jwt::verifier to add additional verification algorithms to.
+	/// \param[in]     _jwk      The JWK containing the required JWA information.
+	/// \param[in]     _alg      The signing algorithm requested by the signed JWT.
+	auto add_asymmetric_alg_from_jwk(
+		jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier,
+		const jwt::jwk<jwt::traits::nlohmann_json>& _jwk,
+		std::string_view _alg) -> void
+	{
+		namespace logging = irods::http::log;
+
+		auto algorithm_family{_alg.substr(0, 2)};
+		auto algorithm_specifics{_alg.substr(2)};
+
+		if (algorithm_family == "RS" || algorithm_family == "PS") {
+			logging::trace(
+				"{}: Detected [{}], attempting extraction of attributes from JWK...", __func__, algorithm_family);
+
+			// Get modulus parameter (JWA Section 6.3.1)
+			auto mod{_jwk.get_jwk_claim("n").as_string()};
+
+			// Get exponent parameter (JWA Section 6.3.1)
+			auto exp{_jwk.get_jwk_claim("e").as_string()};
+
+			// Create public key
+			auto pub_key{jwt::helper::create_public_key_from_rsa_components(mod, exp)};
+
+			// Add verification algorithm
+			if (algorithm_family == "RS") {
+				if (algorithm_specifics == "256") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::rs256(pub_key));
+				}
+				else if (algorithm_specifics == "384") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::rs384(pub_key));
+				}
+				else if (algorithm_specifics == "512") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::rs512(pub_key));
+				}
+			}
+			// "PS"
+			else {
+				if (algorithm_specifics == "256") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps256(pub_key));
+				}
+				else if (algorithm_specifics == "384") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps384(pub_key));
+				}
+				else if (algorithm_specifics == "512") {
+					logging::trace("{}: Adding [{}] to allowed verification algorithms.", __func__, _alg);
+					_verifier.allow_algorithm(jwt::algorithm::ps512(pub_key));
+				}
+			}
+			return;
```

Comment:
Same thing here as above. The `else` cases above are unhandled and this will not fall through to the log statement below.

Reply:
Done.